boards/components/src/print_log.rs

```diff
@@ -0,0 +1,84 @@
+//! Components for PrintLog, a system call driver that outputs
+//! userspace printf statements into the debug output buffer of the
+//! kernel; this has the property that userspace and kernel print
+//! statements are temporally ordered (unlike in Console, where they
+//! can be reordered).
+//!
+//! This provides one Component, `PrintComponent`, which which
+//! implements the same printing system call API as Console but puts
+//! strings into the debug queue.  Prints from userspace that are too
+//! long for the queue are truncated.
+//!
+//! Usage
+//! -----
+//! ```rust
+//! let ast = &sam4l::ast::AST;
+//! let mux_alarm = components::alarm::AlarmMuxComponent::new(ast)
+//!     .finalize(components::alarm_mux_component_static!(sam4l::ast::Ast));
+//! ast.configure(mux_alarm);
+//! let print_log = components::print_log::PrintLogComponent::new(board_kernel, driver_num, mux_alarm)
+//!     .finalize(components::printlog_component_static!(sam4l::ast::Ast));
+
+//! ```
+// Author: Philip Levis <pal@cs.stanford.edu>
+// Last modified: 1/08/2020
```

Comment:
```suggestion
// Last modified: 11/29/2022
```
I realize these always end up out of date, but at least the first one should be right!

Reply:
:)